File: TripKitAndroid/src/main/java/com/skedgo/android/tripkit/alerts/ModeInfo.java
package com.skedgo.android.tripkit.alerts;

import android.support.annotation.Nullable;

import com.google.gson.annotations.JsonAdapter;

import org.immutables.gson.Gson;
import org.immutables.value.Value;

import skedgo.tripkit.routing.ServiceColor;

@Value.Immutable
@Gson.TypeAdapters
@JsonAdapter(GsonAdaptersModeInfo.class)
public interface ModeInfo {
  @Nullable ServiceColor color();
  @Nullable String identifier();
  String modeIdentifier();
}

File: TripKitAndroid/src/main/java/com/skedgo/tripkit/alerts/ModeInfo.kt
package com.skedgo.tripkit.alerts

import com.google.gson.annotations.JsonAdapter
import com.skedgo.tripkit.routing.ServiceColor
import org.immutables.gson.Gson.TypeAdapters
import org.immutables.value.Value.Immutable
import org.immutables.value.Value.Style

@Immutable
@TypeAdapters
@Style(passAnnotations = [JsonAdapter::class])
@JsonAdapter(
    GsonAdaptersModeInfo::class
)
interface ModeInfo {
    fun color(): ServiceColor?

    fun identifier(): String?

    fun alt(): String?
}